src/briefcase/config.py

```diff
@@ -602,30 +665,32 @@ def parse_config(config_file, platform, output_format, console):
         raise BriefcaseConfigError("No Briefcase apps defined in pyproject.toml") from e
 
     for name, config in [("project", global_config)] + list(all_apps.items()):
+        if pep639_info := get_pep639_license_info(config, console, cwd=cwd):
+            config["license"] = pep639_info
+
```

Comment:
This is at least partially a pre-existing problem - but should this validation block be at *after* the app/platform/format merge? At least in theory, a user could put license definitions at every level of the file, and we should be incorporating all that data into the final evaluation of whether the license definition is valid.